cluster/test_cluster.go

```diff
@@ -0,0 +1,136 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"crypto/ecdsa"
+	crand "crypto/rand"
+	"fmt"
+	"io"
+	"math/rand"
+	"net"
+	"testing"
+
+	"github.com/coinbase/kryptology/pkg/signatures/bls/bls_sig"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	"github.com/stretchr/testify/require"
+
+	"github.com/obolnetwork/charon/p2p"
+	"github.com/obolnetwork/charon/tbls"
+	"github.com/obolnetwork/charon/testutil"
+)
+
+// NewForT returns a new cluster lock with dv number of distributed validators, k threshold and n peers.
+// It also returns the peer p2p keys and BLS secret shares. If the seed is zero a random cluster on available loopback
+// ports is generated, else a deterministic cluster is generated.
+// Note this is not defined in testutil since it is tightly coupled with the cluster package.
+func NewForT(t *testing.T, dv, k, n, seed int) (Lock, []*ecdsa.PrivateKey, [][]*bls_sig.SecretKeyShare) {
```

Comment:
shouldn't we name it something like NewCLockForT or something specific?

Reply:
For me the problem is not not saying Cluster, as if this is the only thing this package contains we will use it as cluster.New. Foir me the problem is using T for Test, as it is not a common abbreviation

Reply:
it might not be common, but it is clear, `NewForT(t testing.T, ...)` is very obvious. Don't be scared to innovate 😉